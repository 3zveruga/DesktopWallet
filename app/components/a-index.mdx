---
name: Home
route: /
---

# UI Components

Use the sidebar menu to browse the styleguide of React components available for the Zepio application.
