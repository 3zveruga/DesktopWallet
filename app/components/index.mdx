---
name: Home
route: /
---

# UI Components

Use the sidebar menu to browse the styleguide
