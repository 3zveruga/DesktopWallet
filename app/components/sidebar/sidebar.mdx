---
name: Sidebar
---

import { Playground, PropsTable } from 'docz'
import { SidebarComponent } from './index.js'

# Sidebar

<PropsTable of={SidebarComponent} />

## Basic usage

<Playground>
  <SidebarComponent />
</Playground>
