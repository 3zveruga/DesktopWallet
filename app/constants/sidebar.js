// @flow
import DashboardIcon from '../assets/images/dashboard_icon.svg';
import DashboardIconActive from '../assets/images/dashboard_icon_active.svg';
import ConsoleIcon from '../assets/images/console_icon.svg';
import ConsoleIconActive from '../assets/images/console_icon_active.svg';
import SendIcon from '../assets/images/send_icon.svg';
import SendIconActive from '../assets/images/send_icon_active.svg';
import ReceiveIcon from '../assets/images/receive_icon.svg';
import ReceiveIconActive from '../assets/images/receive_icon_active.svg';
import TransactionsIcon from '../assets/images/transactions_icon.svg';
import TransactionsIconActive from '../assets/images/transactions_icon_active.svg';
import SettingsIcon from '../assets/images/settings_icon.svg';
import SettingsIconActive from '../assets/images/settings_icon_active.svg';

import {
  DASHBOARD_ROUTE,
  SEND_ROUTE,
  RECEIVE_ROUTE,
  SETTINGS_ROUTE,
  CONSOLE_ROUTE,
  TRANSACTIONS_ROUTE,
} from './routes';

export const MENU_OPTIONS = [
  {
    label: 'Dashboard',
    route: DASHBOARD_ROUTE,
    icon: (isActive: boolean) => (isActive ? DashboardIconActive : DashboardIcon),
  },
  {
    label: 'Send',
    route: SEND_ROUTE,
    icon: (isActive: boolean) => (isActive ? SendIconActive : SendIcon),
  },
  {
    label: 'Receive',
    route: RECEIVE_ROUTE,
    icon: (isActive: boolean) => (isActive ? ReceiveIconActive : ReceiveIcon),
  },
  {
    label: 'Transactions',
    route: TRANSACTIONS_ROUTE,
    icon: (isActive: boolean) => (isActive ? TransactionsIconActive : TransactionsIcon),
  },
  {
    label: 'Settings',
    route: SETTINGS_ROUTE,
    icon: (isActive: boolean) => (isActive ? SettingsIconActive : SettingsIcon),
  },
  {
    label: 'Console',
    route: CONSOLE_ROUTE,
    icon: (isActive: boolean) => (isActive ? ConsoleIconActive : ConsoleIcon),
  },
];
