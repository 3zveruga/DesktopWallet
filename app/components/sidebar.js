// @flow

/* eslint-disable max-len */
import React from 'react';
import styled from 'styled-components';
import type { Location, RouterHistory } from 'react-router-dom';
import { MENU_OPTIONS } from '../constants/sidebar';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: ${props => props.theme.sidebarWidth};
  height: ${props => `calc(100vh - ${props.theme.headerHeight})`};
  font-family: ${props => props.theme.fontFamily};
  background-color: ${props => props.theme.colors.sidebarBg};
  padding-top: 15px;
  position: relative;
`;

/* eslint-disable max-len */
type StyledLinkProps = PropsWithTheme<{ isActive: boolean }>;
const StyledLink = styled.a`
  color: ${(props: StyledLinkProps) => (props.isActive ? props.theme.colors.sidebarItemActive : props.theme.colors.sidebarItem)};
  font-size: ${(props: StyledLinkProps) => `${props.theme.fontSize.regular}em`};
  text-decoration: none;
  font-weight: ${(props: StyledLinkProps) => String(props.theme.fontWeight.bold)};
  background-color: ${(props: StyledLinkProps) => (props.isActive ? `${props.theme.colors.sidebarHoveredItem}` : 'transparent')};
  letter-spacing: 0.25px;
  padding: 25px 20px;
  height: 35px;
  width: 100%;
  display: flex;
  align-items: center;
  outline: none;
  border-right: ${(props: StyledLinkProps) => (props.isActive ? `3px solid ${props.theme.colors.sidebarItemActive}` : 'none')};
  cursor: pointer;
  transition: all 0.03s ${(props: StyledLinkProps) => props.theme.transitionEase};

  &:hover {
    color: ${(props: StyledLinkProps) => (props.isActive ? props.theme.colors.sidebarItemActive : '#ddd')}
    background-color: ${(props: StyledLinkProps) => props.theme.colors.sidebarHoveredItem};
  }
`;

const Icon = styled.img`
  width: 17px;
  height: 17px;
  margin-right: 13px;

  ${StyledLink}:hover & {
    filter: ${(props: StyledLinkProps) => (props.isActive ? 'none' : 'brightness(300%)')};
  }
`;

type MenuItem = {
  route: string,
  label: string,
  icon: (isActive: boolean) => string,
};

type Props = {
  history: RouterHistory,
  options?: MenuItem[],
  location: Location,
};

export const SidebarComponent = ({ options, location, history }: Props) => (
  <Wrapper id='sidebar'>
    {(options || []).map((item) => {
      const isActive = location.pathname === item.route;

      return (
        <StyledLink
          isActive={isActive}
          key={item.route}
          onClick={() => (isActive ? {} : history.push(item.route))}
        >
          <Icon isActive={isActive} src={item.icon(isActive)} Alt={`${item.route}`} />
          {item.label}
        </StyledLink>
      );
    })}
  </Wrapper>
);

SidebarComponent.defaultProps = {
  options: MENU_OPTIONS,
};
