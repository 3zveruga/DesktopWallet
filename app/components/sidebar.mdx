---
name: Sidebar
---

import { Playground, PropsTable } from 'docz'

import { SidebarComponent } from './sidebar.js'
import { DoczWrapper } from '../theme.js'

# Sidebar

## General

Sidebar component does not behave correctly inside Docz Styleguide as it is tighly coupled with the router.

## Properties

<PropsTable of={SidebarComponent} />

## Basic Usage

<Playground>
  <DoczWrapper>
    {() => <SidebarComponent location={{}} />}
  </DoczWrapper>
</Playground>
