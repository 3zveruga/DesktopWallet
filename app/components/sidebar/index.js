// @flow

import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { MENU_OPTIONS } from '../../constants/sidebar';
import { styles } from './styles';

// TODO: Not sure this is the best approach to styling
// in a StyledComponents-powered application
const Wrapper = styled.div`
  ${styles.wrapper}
`;

type MenuItem = { route: string, label: string };

type Props = {
  options?: MenuItem[],
};

export const SidebarComponent = ({ options }: Props) => (
  <Wrapper>
    {(options || []).map(item => (
      <Link key={item.route} to={item.route}>
        {item.label}
      </Link>
    ))}
  </Wrapper>
);

SidebarComponent.defaultProps = {
  options: MENU_OPTIONS,
};
