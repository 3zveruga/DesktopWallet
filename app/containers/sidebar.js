// @flow

import { SidebarComponent } from '../components/sidebar';

export const SidebarContainer = SidebarComponent;
