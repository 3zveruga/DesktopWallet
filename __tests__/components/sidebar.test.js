// @flow

import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import 'jest-dom/extend-expect';

import { SidebarComponent } from '../../app/components/sidebar';
import appTheme from '../../app/theme';

afterEach(cleanup);

describe('<SidebarComponent />', () => {
  describe('render()', () => {
    test('should render correctly', () => {
      // $FlowFixMe
      const { asFragment } = render(
        <ThemeProvider theme={appTheme}>
          <MemoryRouter>
            <SidebarComponent location={{ pathname: '/', hash: '/', search: '' }} />
          </MemoryRouter>
        </ThemeProvider>,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
