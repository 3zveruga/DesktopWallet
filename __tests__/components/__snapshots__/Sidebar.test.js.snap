// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarComponent /> render() should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa igFjJZ"
  >
    <a
      class="sc-bwzfXH kBVjXq"
      href="/"
    >
      <img
        alt="/"
        class="sc-htpNat kwYHDD"
        src="[object Object]"
      />
      Dashboard
    </a>
    <a
      class="sc-bwzfXH cYKDCP"
      href="/send"
    >
      <img
        alt="/send"
        class="sc-htpNat eZYlLl"
        src="[object Object]"
      />
      Send
    </a>
    <a
      class="sc-bwzfXH cYKDCP"
      href="/receive"
    >
      <img
        alt="/receive"
        class="sc-htpNat eZYlLl"
        src="[object Object]"
      />
      Receive
    </a>
    <a
      class="sc-bwzfXH cYKDCP"
      href="/transactions"
    >
      <img
        alt="/transactions"
        class="sc-htpNat eZYlLl"
        src="[object Object]"
      />
      Transactions
    </a>
    <a
      class="sc-bwzfXH cYKDCP"
      href="/settings"
    >
      <img
        alt="/settings"
        class="sc-htpNat eZYlLl"
        src="[object Object]"
      />
      Settings
    </a>
    <a
      class="sc-bwzfXH cYKDCP"
      href="/console"
    >
      <img
        alt="/console"
        class="sc-htpNat eZYlLl"
        src="[object Object]"
      />
      Console
    </a>
  </div>
</DocumentFragment>
`;
